Hippo.Web/ClientApp/src/components/Order/CreateOrder.tsx

```diff
@@ -0,0 +1,175 @@
+          //   .reduce((acc, payment) => acc + parseFloat(payment.amount), 0);
+          // setBalancePending(balancePending);
+        } else {
+          alert("Error fetching product for order");
+        }
+      };
+
+      fetchProductOrder();
+    } else {
+      if (isClusterAdmin === false) {
+        window.location.href = `/${cluster}/product/index`;
+      } else {
+        setOrder(defaultOrder);
+      }
+    }
+  }, [cluster, isClusterAdmin, productId]);
+
+  // async function so the form can manage the loading state
+  const submitOrder = async (updatedOrder: OrderModel) => {
+    const editedOrder: OrderModel = {
+      // uneditable fields
+      id: updatedOrder.id,
+      status: updatedOrder.status,
+      createdOn: updatedOrder.createdOn,
+      total: updatedOrder.total,
+      subTotal: updatedOrder.subTotal,
+      balanceRemaining: updatedOrder.balanceRemaining,
+      payments: updatedOrder.payments,
+      history: updatedOrder.history,
+      piUser: updatedOrder.piUser,
+      percentTotal: updatedOrder.percentTotal,
+      nextPaymentDate: updatedOrder.nextPaymentDate,
+
+      // editable fields
+      PILookup: updatedOrder.PILookup,
+      name: updatedOrder.name,
+      productName: updatedOrder.productName,
+      description: updatedOrder.description,
+      category: updatedOrder.category,
+      externalReference: updatedOrder.externalReference,
+      notes: updatedOrder.notes,
+      units: updatedOrder.units,
+      unitPrice: updatedOrder.unitPrice,
+      quantity: updatedOrder.quantity,
+      installments: updatedOrder.installments,
+      installmentType: updatedOrder.installmentType,
+      adjustment: updatedOrder.adjustment,
+      adjustmentReason: updatedOrder.adjustmentReason,
+      adminNotes: updatedOrder.adminNotes,
+      metaData: updatedOrder.metaData,
+      lifeCycle: updatedOrder.lifeCycle,
+      expirationDate: updatedOrder.expirationDate,
+      installmentDate: updatedOrder.installmentDate,
+      billings: updatedOrder.billings,
+    };
+
+    console.log(editedOrder);
+    console.log(JSON.stringify(editedOrder));
+
+    const req = authenticatedFetch(`/api/${cluster}/order/Save`, {
+      method: "POST",
+      body: JSON.stringify(editedOrder),
+    });
+
+    setNotification(req, "Saving", "Order Saved", async (r) => {
+      if (r.status === 400) {
+        const errors = await parseBadRequest(response);
+        return errors;
+      } else {
+        return "An error happened, please try again.";
+      }
+    });
+
+    const response = await req;
+
+    if (response.ok) {
+      const data = await response.json();
+
+      window.location.href = `/${cluster}/order/details/${data.id}`;
+    }
+
+    setOrder(editedOrder); // should be newOrder once it's pulling from the API
+  };
+
+  if (isClusterAdmin === null) {
+    return <div>Loading...</div>;
+  }
+
+  if (!order) {
+    return <div>Loading... {productId}</div>;
+  }
+
```

Comment:
I'd rather bounce them back to the products page if they fiddle with the URL

Reply:
you can still do that (and the effect will redirect them). i just like to handle potential error states both when fetching and when rendering. in your rendering logic right now, the case where `isClusterAdmin === false` isn't handled, so there's not fallback for what is actually displayed. it's not a big deal, especially in this case since i doubt we'll ever change the redirect, but it's a good habit to get into: both "how do i handle errors inside a function" and "what is rendered when there are errors". prob would only happen if something super weird happens, like the effect fails. but debugging stuff like that is a pain and having something specific rendered in that case is nice

Reply:
#396